### PR TITLE
fix: remove redundant blocklyNonSelectable class and integrate non-selectability into existing classes #8328

### PR DESCRIPTION
**Pull request by dakshkanaujia**
## The Basics

- [x] I [validate my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

## The Details

The pull request removes redundant `blocklyNonSelectable` class and integrated non-selectability into existing and more generic classes

## Resolves
Fixes [#8328](https://github.com/google/blockly/issues/8328)

### CODE REVIEW

**Review by BeksOmega — CHANGES_REQUESTED** [review]
Hello @dakshkanaujia It looks like you made some extra changes (to a .gitpod file and a blocklyMenu.code-search file) Could you remove those?

File: core/toolbox/toolbox.ts [hunk 1]
```diff
@@ -197,7 +197,8 @@ export class Toolbox
     const toolboxContainer = document.createElement('div');
     toolboxContainer.setAttribute('layout', this.isHorizontal() ? 'h' : 'v');
     dom.addClass(toolboxContainer, 'blocklyToolboxDiv');
-    dom.addClass(toolboxContainer, 'blocklyNonSelectable');
+    // blocklyNonSelectable was redundant, its functionality was transferred to more relevant classes
+    // dom.addClass(toolboxContainer, 'blocklyNonSelectable');
```
Comment (BeksOmega) [inline core/toolbox/toolbox.ts:201]: Can you remove this instead of commenting it out?

File: core/toolbox/separator.ts [hunk 2]
```diff
@@ -89,6 +89,7 @@ Css.register(`
 }
 
 .blocklyToolboxDiv[layout="h"] .blocklyTreeSeparator {
+  user-select: none;
```
Comment (BeksOmega) [inline core/toolbox/separator.ts:92]: I understand where all of the other changes came from, but why is this here?

**Review by BeksOmega — COMMENTED** [review]
One question, then I'll let CI run and get this merged =)

**Comment by dakshkanaujia** [comment]
Hey @BeksOmega , Thank you for pointing that out =), I have done the required changes in the latest PR, please review it !

**Comment by dakshkanaujia** [comment]
Hopefully! now the checks will pass

**Comment by BeksOmega** [comment]
Ah sorry, it looks like my comment didn't post.

This needs to be against the `rc/v12.0.0` instead of develop. To rebase onto the rc/v12.0.0 branch, you can follow these steps.


1. **Fetch the latest changes from `rc/v12.0.0`:**

   ```bash
   git fetch upstream rc/v12.0.0
   ```

2. **Checkout your branch:**

   ```bash
   git checkout [your-branch-name]
   ```

3. **Start an interactive rebase:**

   ```bash
   git rebase -i upstream/rc/v12.0.0
   ```

4. **In the editor that opens, identify the commits that aren't yours.** These are likely the ones at the top of the list. 

5. **Change the word `pick` to `drop` for each commit you want to remove.** 

6. **Save and close the editor.**

8. **Force push your changes to your branch:** Note that force pushing is a dangerous operation because it overwrites commit history, so if you dropped the wrong commits you could lose work.

   ```bash
   git push -f origin [your-branch-name]
   ```

Once you've done this, the PR should be on the correct branch.

Let me know if you have any questions or run into any issues!

**Comment by dakshkanaujia** [comment]
@BeksOmega , done with rebasing, please review!

**Comment by dakshkanaujia** [comment]
If this is still incorrect , i have raised a new PR , on the `rc/v12.0.0` branch
[https://github.com/google/blockly/pull/8433](New_PR_LINK)

**Pull request closed without merging**